pdfgen.c

```diff
@@ -2511,26 +2567,45 @@ int pdf_add_rgb24(struct pdf_doc *pdf, struct pdf_object *page, float x,
     return pdf_add_image(pdf, page, obj, x, y, display_width, display_height);
 }
 
+int pdf_add_grayscale8(struct pdf_doc *pdf, struct pdf_object *page, float x,
+                       float y, float display_width, float display_height,
+                       const uint8_t *data, uint32_t width, uint32_t height)
+{
+    struct pdf_object *obj;
+
+    obj = pdf_add_raw_grayscale8(pdf, data, width, height);
+    if (!obj)
+        return pdf->errval;
+
+    return pdf_add_image(pdf, page, obj, x, y, display_width, display_height);
```

Comment:
Does this now need a call to `get_img_display_dimensions`?

Reply:
yeah, this PR branch needs to be rebased from master.

I'd have done this already but my laptop's cpu cooler has decided to die on me so I think I'll get back to this in a couple of days.

Reply:
Let me know if you'd like me to take this over and clean up the last small bits for merge. I think it's almost sorted.

Reply:
Update: yeah it will probably take some time until I can get back to this, if you could finish this up that would be great.

Reply:
No problem. I also realised that your new greyscale support lets us trivially add in support for .pgm binary image files as well, so I've done that. That also means that the fuzz tester will be able to reach the new greyscale functions, which is also nice.

Reply:
Yet another format, that's nice to see.
Thank you for finishing this up.